### ISSUE TITLE
Whether to use 'logits' or 'loss' in LabelSmoother

### ISSUE THREAD

**Issue by zalmanchen** [huggingface/transformers#20567]
### System Info

- 'transformers' version: 4.24.0

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

it is in "transformers/trainer.py"
![ec8d50fbf9388598e5bb7a631f406f364921130857e902ff2d73b9dff332d30cQzpcVXNlcnNcemFsbWFcQXBwRGF0YVxSb2FtaW5nXERpbmdUYWxrXDQ3NDQyMTE1Ml92MlxJbWFnZUZpbGVzXDE2NzAwODU4NjkwOThfREVGN0M0QzMtRjI1Mi00Y2UyLUE2RDYtNjFCQzI4MENEOTZDLnBuZw==](https://user-images.githubusercontent.com/70104951/205451920-7b0c4a21-7b83-4873-b04a-e9ba6ba97e03.png)

it is in "transformers/trainer_pt_utils.py"

![75a7ca0c9a5623cd394ed71be3651c998b0d15e9238e67fe612854c05af75025QzpcVXNlcnNcemFsbWFcQXBwRGF0YVxSb2FtaW5nXERpbmdUYWxrXDQ3NDQyMTE1Ml92MlxJbWFnZUZpbGVzXDE2NzAwNTY0NTI2MTdfODUyRkFENTUtRDVDRi00YTE4LUIyNTItRUJENjg0MTU3MUNFLnBuZw==](https://user-images.githubusercontent.com/70104951/205450770-bca21a08-0115-4895-94ae-f7cfd76d6e61.png)

### Expected behavior

In fact, it is not a bug problem! I just have doubts about the variable used in a function

In "transformers/trainer_pt_utils.py", the class "LabelSmoother \__call\__" function, I noticed that it uses the **'logits'** value, but **'output [0]'** is selected under 'else' condition, if I am not mistaken, **'output[0]'** should represent loss(I use the **_BartForConditionalGeneration_**) . So is there any problem here?
